### PR TITLE
Skip Topological sorting if not required

### PR DESCRIPTION
**Pull request by bahusoid**

### CODE REVIEW

**Comment by bahusoid** [comment]
Hm. It seems there is no need to split dependent alias calculation and sorting across different methods. I'm going to remove `_dependentAliases` field

File: src/NHibernate/Loader/JoinWalker.cs [hunk 1]
```diff
@@ -840,6 +851,21 @@ protected JoinFragment MergeOuterJoins(IList<OuterJoinableAssociation> associati
 			return outerjoin;
 		}
 
+		private IList<OuterJoinableAssociation> GetSortedAssociations(IList<OuterJoinableAssociation> associations)
+		{
+			var indexes = GetTopologicalSortOrder(associations);
+			if (indexes == null)
```
Comment (hazzik) [inline src/NHibernate/Loader/JoinWalker.cs]: Can we remove null-driven logic here? Eg move conditions and dependents calculation from `GetTopologicalSortOrder` to here.

File: src/NHibernate/Loader/JoinWalker.cs [hunk 2]
```diff
@@ -840,6 +851,21 @@ protected JoinFragment MergeOuterJoins(IList<OuterJoinableAssociation> associati
 			return outerjoin;
 		}
 
+		private IList<OuterJoinableAssociation> GetSortedAssociations(IList<OuterJoinableAssociation> associations)
```
Comment (hazzik) [inline src/NHibernate/Loader/JoinWalker.cs]: This can be made static.

**Comment by hazzik** [comment]
Can you also review #419?

**Comment by bahusoid** [comment]
Regarding #419. I know why and what this sorting does. But I have no idea how :) So I can't really say if it's any better then existing algorithm.

File: src/NHibernate/Loader/JoinWalker.cs (hunk 1 above)
Comment (bahusoid) [reply src/NHibernate/Loader/JoinWalker.cs]: Done

**Comment by bahusoid** [comment]
Something is fishy with MergeDeliveryNodeAsync test. It's failed to often on appveyor and takes  7 min when failed

**Pull request merged**